src/handlers/rustc_commits.rs

```diff
@@ -70,19 +72,30 @@ pub async fn handle(ctx: &Context, event: &Event) -> anyhow::Result<()> {
         return Ok(());
     }
 
-    let mut sha = bors.merge_sha;
-    let mut pr = Some(event.issue.number.try_into().unwrap());
+    synchronize_commits(ctx, &bors.merge_sha, event.issue.number.try_into().unwrap()).await;
 
+    Ok(())
+}
+
+/// Fetch commits that are not present in the database.
+async fn synchronize_commits(ctx: &Context, sha: &str, pr: u32) {
     let db = ctx.db.get().await;
-    loop {
-        // FIXME: ideally we would pull in all the commits here, but unfortunately
-        // in rust-lang/rust's case there's bors-authored commits that aren't
-        // actually from rust-lang/rust as they were merged into the clippy repo.
+    let mut pr = Some(pr);
+
+    // List of roots to be resolved. Each root and its parents will be recursively resolved
+    // until an existing commit is found.
+    let mut to_be_resolved = VecDeque::new();
+    to_be_resolved.push_back(sha.to_string());
+    to_be_resolved.extend(get_missing_commits(&db).await);
+
+    while !to_be_resolved.is_empty() {
+        let sha = to_be_resolved.pop_front().unwrap();
```

Comment:
`while let Some(sha) = to_be_resolved.pop_front()` probably is slightly nicer here.

I also don't know that we need a queue vs. just a simple Vec, but either seems OK.